sdk/tests/conformance/textures/misc/video-rotation.html

```diff
@@ -0,0 +1,114 @@
+<!--
+Copyright (c) 2020 The Khronos Group Inc.
+Use of this source code is governed by an MIT-style license that can be
+found in the LICENSE.txt file.
+-->
+
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <title>Verifies EXIF orientation is respected when uploading images to WebGL textures</title>
+    <link rel="stylesheet" href="../../../resources/js-test-style.css" />
+    <script src="../../../js/js-test-pre.js"></script>
+    <script src="../../../js/webgl-test-utils.js"></script>
+    <script src="../../../js/tests/tex-image-and-sub-image-utils.js"></script>
+</head>
+
+<body onload="run()">
+    <canvas id="c" width="256" height="256"></canvas>
+    <div id="description"></div>
+    <div id="console"></div>
+    <script>
+        "use strict";
+        description();
+        let wtu = WebGLTestUtils;
+        let tiu = TexImageUtils;
+        let canvas = document.getElementById("c");
+        let gl = wtu.create3DContext(canvas);
+        let program = tiu.setupTexturedQuad(gl, gl.RGBA);
+        const resourcePath = "../../../resources/";
+        const tolerance = 5;
+
+        const expectedColors = {
+            top: { location: [0.5, 0.25], color: [255, 0, 0] },
+            left: { location: [0.4, 0.5], color: [0, 0, 255] },
+            right: { location: [0.6, 0.5], color: [255, 255, 0] },
+            bottom: { location: [0.5, 0.75], color: [0, 255, 0] },
+        }
+
+        function output(str) {
+            debug(str);
+            bufferedLogToConsole(str);
+        }
+
+        function checkPixels() {
+            for (let place in expectedColors) {
+                let color = expectedColors[place];
+                let loc = color.location;
+                let x = loc[0];
+                let y = loc[1];
+                output("  Checking " + place);
+                wtu.checkCanvasRect(gl, Math.floor(canvas.width * x), Math.floor(canvas.height * y), 1, 1,
+                    color.color, "shouldBe " + color.color + " +/-" + tolerance, tolerance);
+            }
+        }
+
+        function loadVideoElement(filename) {
+            return new Promise((resolve) => {
+                const video = new HTMLVideoElement();
+                video.addEventListener('loadeddata', () => {
+                   resolve(video);
```

Comment:
Please use WebGLTestUtils.startPlayingAndWaitForVideo. That function already contains several improvements to make videos behave more portably between browsers.